### ISSUE TITLE
No type check in setting weight function of CBox

### ISSUE THREAD

**Issue by gtaifu** [DiCarloLab-Delft/PycQED_py3#24]
@AdriaanRol 
The function `_set_int_weights` used to set the integration weight of the Master Controller.

It should only accept integers ranging from -128 to 127. But it does not have parameter type check for the weights. One consequence is that the floating point integration weights are translated into binary and sent to the CBox. Of course, the binary is not correct.

**Comment by AdriaanRol** [comment]
@NielsBultink @gtaifu
As I'm working with the CBox now I thought I'd address this. A small question: Is it acceptable to enforce a fixed lenght of the integration weights?
 In normal use we do this already, we fix the weights to the max length we can set.

**Comment by gtaifu** [comment]
For me it is fine.

**Comment by AdriaanRol** [comment]
@gtaifu , will you include this in PR #98 ?

**Comment by gtaifu** [comment]
Hi Adriaan, the type and range check of the weights has been included in PR #98 . In terms of the fixed length of the weights, somebody has already done it in the past using the following code:
``` python
        if len(weights) != 512:
            raise ValueError('Length of array must be 512 elements')
```

**Comment by AdriaanRol** [comment]
This was closed when merging #98
